### ISSUE TITLE
Make gstat as an internal command

### ISSUE THREAD

**Issue by otakutyrant** [nushell/nushell#10910]
### Describe the solution you'd like

It is painful to install the gstat plugin manually everytime I deploy my nushell plugin in a new PC.

The gstat plugin is this: https://github.com/nushell/nushell/tree/main/crates/nu_plugin_gstat

**Comment by amtoine** [comment]
@fdncred 
i figure this is one of your plugins originally, right?
what do you think about that?

not sure a wrapper around Git makes too much sense as a built-in command :thinking:

**Comment by fdncred** [comment]
Yup, I wrote `gstat` ages ago. I believe `gstat` was originally a built-in command but we moved it to a plugin because we didn't want to take a dependency on git2. I work on-again/off-again on porting it to `gix`/`gitoxide` since that's what `cargo` is moving to but haven't finished it yet. Gix will probably have more dependencies, but it should work better/faster. But, if enough people want it, I can see us moving forward with moving `gstat` into nushell somewhere.

**Comment by hholst80** [comment]
> It is painful to install the gstat plugin

What is the process required to do that? I think having ease of access to plugins is the way to go.  Look at the Neovim community.  Plugins autoinstall, lazy load, etc.  It's seemless.  Just bring on your own config.

**Comment by fdncred** [comment]
> What is the process required to do that? 

It should be:
1. cargo install nu_plugin_gstat (or you can clone the repo and build/install it yourself)
2. register ~/.cargo/bin/nu_plugin_gstat

**Comment by otakutyrant** [comment]
It seems that the convention is use `plugin add` instead of `register` now.

**Comment by fdncred** [comment]
yup, things change over the years. register has been removed in favor of plugin add and plugin use.

**Comment by sholderbach** [comment]
Can we separate out suggestions how we can make the distribution/installation of plugins simpler from the premise of the issue as stated in the title. Because that I don't see that happening and would vote to close the issue.

**Comment by otakutyrant** [comment]
It seems that the gstat became [internal](https://www.nushell.sh/book/plugins.html#core-plugins) finnaly. However, I have to still plugin add it manually, like `plugin add /usr/bin/nu_plugin_gstat`. Any idea?

**Comment by fdncred** [comment]
@otakutyrant gstat is still a plugin and not an internal command. That link you provided is talking about plugins.

**Comment by otakutyrant** [comment]
Okay, I rethought about the idea. I agree that the plugin about git is unnecessary to become internal.
